src/hotspot/os/linux/os_perf_linux.cpp

```diff
@@ -927,11 +926,12 @@ CPUInformationInterface::CPUInformationInterface() {
 
 bool CPUInformationInterface::initialize() {
   _cpu_info = new CPUInformation();
-  _cpu_info->set_number_of_hardware_threads(VM_Version_Ext::number_of_threads());
-  _cpu_info->set_number_of_cores(VM_Version_Ext::number_of_cores());
-  _cpu_info->set_number_of_sockets(VM_Version_Ext::number_of_sockets());
-  _cpu_info->set_cpu_name(VM_Version_Ext::cpu_name());
-  _cpu_info->set_cpu_description(VM_Version_Ext::cpu_description());
+  VM_Version::initialize_cpu_information();
```

Comment:
I can't figure out when this code will actually get executed in  relation to the VM initialization process and VM_Version's initialization. Can this actually execute before that happens? Or could we assert that it has happened?

Reply:
VM_Version::initialize() is called very early in Threads::create_vm.  This latter VM_Version::initialize_cpu_information is called later when JFR event is emitted.  The reason it was "_ext" was because it is part of JFR only.  It seems that we might be able to consolidate this more now that it's moved later.
I don't think adding an assert would be meaningful here.

Reply:
I did add a simple initialization assert in the x86 code where it might be interesting and reran the JFR tests.